dist/icon-button/ds6/icon-button.css

```diff
@@ -82,6 +110,7 @@ a.icon-link--badged .badge {
   display: block;
   height: 24px;
   left: 16px;
+  line-height: 16px;
```

Comment:
Thinking about it, vertically centering with line-height should be fine in this situation, so I take back what I said earlier. You might have better success though if it matches the height of the badge. You might be able to do away with the `padding-top` too.